File: C++/M01_ProgrammingBasics/L05_WhileLoop/Lab/Solutions/P09_Moving.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {

	int width, len, height;
	string command;
	cin >> width >> len >> height >> command;

	int capacity = width * len * height;
	int boxes = 0;
	while (command != "Done") {
		int currBox = stoi(command);
		boxes += currBox;

		if (capacity < boxes) {
			cout << "No more free space! You need " << boxes - capacity << " Cubic meters more." << endl;

			return 0;
		}
		cin >> command;
	}

	cout << capacity - boxes << " Cubic meters left." << endl;

	return 0;
}

File: Python/M01_ProgrammingBasics/L05_WhileLoop/Lab/Solutions/P09_Moving.py
width = int(input())
length = int(input())
height = int(input())

free_space = width * length * height
space_needed = 0
diff = 0

while True:
    line = input()
    if line == "Done":
        break
    boxes = int(line)
    space_needed += boxes
    if space_needed > free_space:
        break

diff = abs(free_space - space_needed)

if space_needed > free_space:
    print(f"No more free space! You need {diff} Cubic meters more.")
else:
    print(f"{diff} Cubic meters left.")